lib/credentials_from_file.c

```diff
@@ -297,179 +194,114 @@ bool read_normal_device_credentials(const char *dev_cred_file,
 		goto end;
 	}
 
-	our_dev_cred->owner_blk->guid = sdo_byte_array_alloc(guid_length);
-	if (!our_dev_cred->owner_blk->guid) {
-		LOG(LOG_ERROR, "Alloc failed\n");
+	if (sdo_blob_read((char *)dev_cred_file, flags, sdor->b.block,
+			  sdor->b.block_size) == -1) {
+		LOG(LOG_ERROR, "Failed to read DeviceCredential blob : Normal.blob\n");
 		goto end;
 	}
 
-	if (!sdor_byte_string(sdor, our_dev_cred->owner_blk->guid->bytes,
-		our_dev_cred->owner_blk->guid->byte_sz)) {
-		LOG(LOG_ERROR, "Element=DCGuid not found\n");
+	if (!sdor_parser_init(sdor)) {
+		LOG(LOG_ERROR, "SDOR Parser Initialization failed!\n");
 		goto end;
 	}
 
-	our_dev_cred->owner_blk->rvlst = sdo_rendezvous_list_alloc();
-	if (!our_dev_cred->owner_blk->rvlst ||
-	    !sdo_rendezvous_list_read(sdor, our_dev_cred->owner_blk->rvlst)) {
-		LOG(LOG_ERROR, "Own's rvlist read Error\n");
+	if (!sdor_start_array(sdor)) {
+		LOG(LOG_ERROR, "DeviceCredential read: Begin Array not found\n");
 		goto end;
 	}
 
-	our_dev_cred->owner_blk->pkh =
-	    sdo_hash_alloc(SDO_CRYPTO_HASH_TYPE_USED, SDO_SHA_DIGEST_SIZE_USED);
-	if (!our_dev_cred->owner_blk->pkh ||
-	    !sdo_hash_read(sdor, our_dev_cred->owner_blk->pkh)) {
-		LOG(LOG_ERROR, "Own's pkh read Error\n");
+	if (!sdor_signed_int(sdor, &our_dev_cred->ST)) {
+		LOG(LOG_ERROR, "DeviceCredential read: ST not found\n");
 		goto end;
 	}
 
-	if (!sdor_end_array(sdor)) {
-		LOG(LOG_ERROR, "End object not found\n");
+	if (our_dev_cred->ST < SDO_DEVICE_STATE_READY1) {
 		goto end;
 	}
 
-	ret = true;
-
-end:
-	if (sdob->block) {
-		sdo_free(sdob->block);
-		sdob->block = NULL;
-	}
-	return ret;
-}
-
-/**
- * Read the Device Credentials blob, contains our MFG Blk
- * @param dev_cred_file - the blob the credentials are saved in
- * @param flags - descriptor telling type of file
- * @param our_dev_cred - pointer to the device credentials block,
- * @return true if read and parsed correctly, otherwise false.
- */
-bool read_mfg_device_credentials(const char *dev_cred_file,
-				 sdo_sdk_blob_flags flags,
-				 sdo_dev_cred_t *our_dev_cred)
-{
-	bool ret = false;
-	size_t dev_cred_len = 0;
-	sdor_t sdoreader = {0};
-	sdor_t *sdor = NULL;
-	sdo_block_t *sdob = NULL;
-
-	sdor = &sdoreader;
-	sdob = &(sdor->b);
-
-	if (!our_dev_cred) {
+	if (!sdor_boolean(sdor, &our_dev_cred->dc_active)) {
+		LOG(LOG_ERROR, "DeviceCredential read: DCActive not found\n");
 		goto end;
 	}
 
-	if (!sdor_init(sdor)) {
-		LOG(LOG_ERROR, "sdor_init() failed!\n");
+	if (!sdor_signed_int(sdor, &our_dev_cred->owner_blk->pv)) {
+		LOG(LOG_ERROR, "DeviceCredential read: DCProtVer not found\n");
 		goto end;
 	}
 
-	dev_cred_len = sdo_blob_size((char *)dev_cred_file, flags);
-	if (dev_cred_len > 0) {
-		// Resize sdob block size
-		sdo_resize_block(sdob, dev_cred_len);
-	} else {
-		LOG(LOG_ERROR, "Could not get %s!\n", (char *)dev_cred_file);
+	size_t device_info_length = 0;
+	if (!sdor_string_length(sdor, &device_info_length) || device_info_length <= 0) {
```

Comment:
size_t is unsigned, comparison <=0 is same as ==0

Reply:
Done.